### ISSUE TITLE
Check suddenly started passing despite both `squash!` and `fixup!` commits

### ISSUE THREAD

**Issue by adrianschmidt** [xt0rted/block-autosquash-commits-action#130]
We've been using this action successfully for a while now, and as far as I know, it has always worked just fine for us, but we pretty much only use `fixup!` commits. Today I added two `squash!` commits to a PR, and to my surprise the check passed rather than failing as expected.

UPDATE: I added an empty `fixup!` commit, just to double check that the action would fail, but it didn't. So there may be an issue with the number of commits in the PR (currently at 42) (we don't normally have this many commits in a PR, this is an outlier). We haven't previously had to add the `read` permission, as outlined in the [readme](https://github.com/xt0rted/block-autosquash-commits-action#control-permissions), but I'll try that too, just in case something changed very recently.

UPDATE 2: Nope, still fails with read-permission set explicitly.

Here's a link to the run: https://github.com/Lundalogik/lime-elements/pull/1297/checks?check_run_id=3122228255

As you can see from these screenshots, we're running v2.1.0 of the action, and the two last commits where `squash!` commits:

![image](https://user-images.githubusercontent.com/154725/126463320-7ec1164e-7d91-47f5-818a-40dca65b4cec.png)

![image](https://user-images.githubusercontent.com/154725/126463381-a3a77719-5056-4314-9c5e-263b976577cf.png)

This is the section of the workflow file that concerns this job:

```yaml
name: Pull Request Checks
on: [pull_request]

jobs:

# …snip…

  autosquash:
    name: Block Autosquash Commits
    runs-on: ubuntu-latest
    steps:
      - name: Block Autosquash Commits
        uses: xt0rted/block-autosquash-commits-action@v2.1.0
        with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}

# …snip…
```

**Comment by xt0rted** [comment]
Thanks for the report @adrianschmidt. It looks like the api call used to get the list of commits is only returning the first 30. This can be increased but has a hard limit of 250. That should be good for now though. I'm testing an update and should have a new release shortly.

**Comment by xt0rted** [comment]
This should be fixed in v2.2.0. Thanks for the support.
